examples/nlp/dialogue/preprocess_for_zero_shot_slot_filling.py

```diff
@@ -0,0 +1,199 @@
+        copy_file(args.preprocess_file_path, output_path, "dict.intents.csv")
+
+
+def copy_file(source_path, target_path, filename):
+    global_parameters = "*"
+    filename_with_path = source_path + filename
+
+    if filename_with_path in glob.glob(os.path.join(source_path, global_parameters)):
+        if filename_with_path not in glob.glob(os.path.join(target_path, global_parameters)):
+            shutil.copy(filename_with_path, target_path)
+            print("The file " + filename_with_path + " is copied!")
+        else:
+            print("{} exists in {}".format(filename_with_path, os.path.join(os.path.split(target_path)[-2:])))
+
+
+def from_slot_name_to_slot_id(train, train_slots):
+
+    # manual build dic
+    dict_slots = []
+    dict_slots = ['O', 'B-LOC', 'I-LOC', 'B-MISC', 'I-MISC', 'B-ORG', 'I-ORG', 'B-PER', 'I-PER']
+
+    dict_slots_id = {}
+    for idx, d in enumerate(dict_slots):
+        dict_slots_id[d] = str(idx)
+
+    train_new = []
+    train_slots_new = []
+
+    for idx, sentence in enumerate(train):
+        if idx == 0:
+            train_new.append(sentence)
+        else:
+            new_label = []
+            for label in train_slots[idx - 1].strip().split():
+                new_label.append(dict_slots_id[label])
+
+            train_new.append(sentence + '\t' + '0')  # every intent label set to 0 at this moment
+            train_slots_new.append(' '.join(new_label))
+
+    return train_new, train_slots_new
+
+
+def fix_input_data_by_combine_punctuation_with_word(train, train_slots):
+    train_new = []
+    train_slots_new = []
+
+    for idx, sentence in enumerate(train):
+        if idx == 0:
+            train_new.append(sentence)
+        else:
+            new_sentence = []
+            new_label = []
+            for word, label in zip(sentence.strip().split(), train_slots[idx - 1].strip().split()):
+                if word == '.' or word == ',' or word == '?' or word == '\'s':
+                    if new_sentence:
+                        new_sentence[-1] = new_sentence[-1] + word
+                else:
+                    new_sentence.append(word)
+                    new_label.append(label)
+            if len(new_sentence) != 0:
+                train_new.append(' '.join(new_sentence) + '\t' + sentence.split()[-1])
+                train_slots_new.append(' '.join(new_label))
+
+            else:
+                train_new.append(sentence)
+                train_slots_new.append(train_slots[idx - 1])
+    return train_new, train_slots_new
+
+
+def remove_sentence_slot_label_pair_in_data_if_without_entity(train, train_slots, dataset_name):
+    train_new = []
+    train_slots_new = []
+
+    if dataset_name == 'assistant':
+        label_for_empty_entity = '54'
+    else:
+        label_for_empty_entity = '0'
+
+    for idx, sentence in enumerate(train):
+        if idx == 0:
+            train_new.append(sentence)
+        else:
+            FLAG_with_entity = False
```

Comment:
do not use var names with UPPER and lower case, if it is a variable change to all lower_case , uppercase is reserved for constants